packages/resources/src/rest/config.ts

```diff
@@ -0,0 +1,85 @@
+import fetch from '@dojo/framework/shim/fetch';
+import { ResourceConfig } from './../provider';
+
+export interface RestResourceUrlOptions {
+	origin: string;
+	name: string;
+	id?: string;
+}
+
+export interface RestResourceUrlFunction {
+	(options: RestResourceUrlOptions): string;
+}
+
+export interface RestResourceOperationsConfig {
+	optimistic: boolean;
+	verb: 'GET' | 'POST' | 'PATCH' | 'DELETE' | 'PUT';
+	url: RestResourceUrlFunction;
+}
+
+export interface RestManyResourceOperations {
+	read: false | RestResourceOperationsConfig;
+}
+
+export interface RestResource<S> {
+	origin: string;
+	name: string;
+	idKey?: string;
+	template?(resource: Partial<S>): S;
+	many?: Partial<RestManyResourceOperations>;
+}
+
+const DEFAULT_REST_CONFIG: {
+	many: RestManyResourceOperations;
+} = {
+	many: {
+		read: {
+			optimistic: true,
```

Comment:
Does optimistic make sense as a possible option on reading/fetching resources? Possibly not as we cannot optimistically update the store like we can for update/create/remove